### ISSUE TITLE
TemplateHaskell '$' gets added as a tag

### ISSUE THREAD

**Issue by chreekat** [elaforge/fast-tags#16]
%%% tags %%%

    $	src/Model.hs	54;"	o

%%% src/Model.hs, line 48–54 %%%

    share [ mkPersist sqlSettings
          , mkMigrate "migrateAll"
          , mkDeleteCascade sqlSettings
          , mkDiscussionTypes
          , mkReferences "Comment"
          ]
        $(persistFileWith lowerCaseSettings "config/models")

**Comment by chreekat** [comment]
Actually, the same happens with QuasiQuotes, where '|' becomes a tag from `|]`

Note I consider these pretty unimportant problems. :)

**Comment by elaforge** [comment]
Yeah, it doesn't seem like a big deal because you're unlikely to try to jump to a $.  But it would still be nice to clean up.  Originally I just got tags from type declarations, which was pretty robust, but a while back someone added a large patch that tries to look for tags in all declarations, and I don't totally understand the new stuff.  I took a look at it yesterday, but the same patch also broke the old tests... so I got distracted trying to fix those.

Anyway, if it turns out to be hard to fix, then maybe I could add a flag to go back to only getting tags from type declarations, since that was always good enough for me.
